### PR TITLE
add support for configurable sms backend

### LINKED ISSUE #664: Configurable SMS Providers
Currently we only support SNS https://github.com/coronasafe/care/blob/develop/care/utils/sms/sendSMS.py#L7-L23
It would be good to have it configurable by the deployments

We could use https://github.com/roaldnefs/django-sms and write our own backends for other sms providers
note, this package will require us to make our own backend for sns as well

### PR DESCRIPTION
**Pull request by AnkurPrabhu**
## Proposed Changes

- Have used django-sms and make a custom backend for our existing sns

### Associated Issue

-  closes #664 

## Merge Checklist

- [x] Tests added/fixed
- [ ] Update docs in `/docs`
- [x] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by AnkurPrabhu** [comment]
@sainak ptal, shoud i add tests for other places as well ?

File: Pipfile [hunk 1]
```diff
@@ -45,6 +45,7 @@ requests = "==2.31.0"
 sentry-sdk = "==1.30.0"
 whitenoise = "==6.5.0"
 redis-om = "==0.2.1"
+django-sms = "*"
```
Comment (sainak) [inline Pipfile]: pin to a specific version

File: care/utils/sms/sendSMS.py [hunk 2]
```diff
@@ -21,3 +25,25 @@ def sendSMS(phone_numbers, message, many=False):
         )
         client.publish(PhoneNumber=phone, Message=message)
     return True
+
+
+class CustomSMSBackendSNS(BaseSmsBackend):
+    def send_messages(self, messages: List[Message]):
+        client = boto3.client(
```
Comment (sainak) [inline care/utils/sms/sendSMS.py]: initialize the client in `__init__()`

File: care/utils/sms/sendSMS.py [hunk 3]
```diff
@@ -21,3 +25,25 @@ def sendSMS(phone_numbers, message, many=False):
...
+        for message in messages:
+            for phone_number in message.recipients:
+                message_body = message.body
+
+                try:
+                    mobile_validator(phone_number)
+                except Exception:
+                    continue
+
+                client.publish(PhoneNumber=phone_number, Message=message_body)
```
Comment (sainak) [inline care/utils/sms/sendSMS.py]: handle exceptions

File: config/settings/local.py [hunk 4]
```diff
@@ -48,3 +48,5 @@
 RUNSERVER_PLUS_PRINT_SQL_TRUNCATE = 100000
 
 DISABLE_RATELIMIT = True
+
+SMS_BACKEND = "care.utils.sms.sendSMS.CustomSMSBackendSNS"
```
Comment (sainak) [inline config/settings/local.py]: set `SMS_BACKEND = "care.utils.sms.sendSMS.CustomSMSBackendSNS"` in base settings
and for local and test use console backend

**Review by sainak — CHANGES_REQUESTED** [review]

**Comment by sainak** [comment]
@vigneshhari though the django-sms provides a standard wrapper for sms, the backend themselves have different names for settings, and some backends require client sdks to be installed
also I just noticed that django-sms has a 3BSD license

**Comment by AnkurPrabhu** [comment]
@sainak let me know if the changes made look good to you

**Comment by sainak** [comment]
Hey @AnkurPrabhu thanks for these changes, but we are planning to move the sms backends to a plugin system, once the plugin architecture becomes stable, we will create our own implementation for the sms

cc: @vigneshhari

**Pull request closed without merging**